### ISSUE TITLE
[FEA] NamedAgg in groupby context

### ISSUE THREAD

**Issue by MarcoGorelli** [rapidsai/cudf#15118]
**Is your feature request related to a problem? Please describe.**

please support namedagg in groupby(...).agg

**Describe the solution you'd like**

to be able to write
```python
df.groupby("A").agg(
    b_min=cudf.NamedAgg(column="B", aggfunc="min"),
    c_sum=cudf.NamedAgg(column="C", aggfunc="sum")
)
```

**Describe alternatives you've considered**

**Additional context**

https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.agg.html

**Comment by shwina** [comment]
Thanks, Marco! It's unlikely the team will be able to take on this soon; but until someone does, hopefully the below is a satisfactory workaround for your needs:


```python
In [133]: df = cudf.DataFrame({'a': [1, 1, 1, 2, 2], 'b': [1, 2, 3, 4, 5], 'c': [1, 2, 3, 4, 5]})

In [134]: result = df.groupby('a').agg({'b': ['min', 'max'], 'c': ['sum']})

In [135]: result.columns = ["_".join(names) for names in result.columns.tolist()]

In [136]: result
Out[136]:
   b_min  b_max  c_sum
a
1      1      3      6
2      4      5      9
```

**Comment by Matt711** [comment]
Closed by #16528
